Bottom: 19396bd6e0c989ad382456b4d945ae7286b1e1f5
Top:    d2effc0fadc8275ddb0bcb2e1200ef61b1eac387
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-06 17:34:28 +0000

Workaround crash where contour is passed as NULL to poly_PreContour










---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
